メッセージ投稿機能の非同期通信

yukri1030 opened the pull request:
#  WHAT
js,jbuilderのファイルの作成。jsでa jaxの定義をし、コントローラーでrespond_toで viewを分ける。その後、jbuidlderにデータが格納されて、jsのdoneメソッドで受け取るという一連の流れを実装した。

#  WHY
スムーズなレスポンスのため、非同期通信を実装する必要があった

投稿の連続送信の実装
https://gyazo.com/dab8d3484dab69370ea16b857b04e7bd

空のまま投稿した際の、アラートの表示
https://gyazo.com/1c8ada2d25e80ba688cb83f43e3add8f

画像の投稿
https://gyazo.com/07453e525c89b882ade1eb950f82acae

yukri1030 committed:
first commit

yukri1030 committed:
不要なコメントアウトの削除

yukri1030 committed:
ajaxの内容を記載

ajaxの内容を記載した。messageコントローラーのcreateアクションには送られている

yukri1030 committed:
データを受け取るためのdoneメソッドを追記した

データを受け取るため、doneメソッドなどを実装した

yukri1030 committed:
画像を受け取る変数がなかったため、定義した

画像を受け取る変数の定義

yukri1030 committed:
respond_toでviewを分けた

respond_toでajaxとhtmlでviewを分けた

yukri1030 committed:
インシデントなどを直した

yukri1030 committed:
画像を受け取るためのcssを追加した

yukri1030 committed:
不要なコメントアウトを削除した

yukri1030 committed:
不要なコメントアウトを削除した

yukri1030 committed:
不要なコメントアウトを削除した

yukri1030 committed:
非同期通信の投稿フォームの実装

# WHAT
jsファイルの実装。ajaxによる送信とdoneメソッドによる受け取り。連続送信とアラートの設定。

# WHY
スムーズなレスポンスのための実装。

yukri1030 committed:
日時の適切な表示

# WHAT
日時を適切な表示にするため、strfrimeの設定を行なった。

# WHY
日付を見やすくするため

miyagit commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,52 @@
+      $('.center').animate({scrollTop: $('.center')[0].scrollHeight}, 'fast');
+
+      $('.form__submit').prop('disabled', false).empty("");
+      $('.form__message').val("");
```
form属性のresetメソッドを使いましょう。

yukri1030 replied:
resetだと、以前の投稿が画面に残ってしまいます・・・

TOMARUMARU replied:
resetメソッドの記述が正しくないのではないでしょうか？

yukri1030 replied:
form属性に指定してなかったことが原因でした。ありがとうございます。

miyagit commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,52 @@
+      $('.center').append(html);
+      $('.center').animate({scrollTop: $('.center')[0].scrollHeight}, 'fast');
+
+      $('.form__submit').prop('disabled', false).empty("");
```
empty("")とは何でしょうか？

yukri1030 replied:
実装のミスです。削除します。

yukri1030 committed:
連続送信の機能の実装

# WHAT
連続送信機能の実装。emptyメソッドが不要だったため削除した。valの代わりにresetメソッドを用いようとしたがうまくいかず。

#WHY
レスポンスを早くするため

yukri1030 committed:
連続送信の実装

# WHAT
resetメソッドを用いて、連続送信の実装を行いました

# WHY
よりスムーズな投稿機能実装のため

miyagit commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,52 @@
+      var html = buildHTML(message);
+      $('.center').append(html);
+      $('.center').animate({scrollTop: $('.center')[0].scrollHeight}, 'fast');
+
```
こちらの空行削除しましょう。

yukri1030 replied:
削除しました

miyagit commented on app/views/messages/_message.html.haml:
```diff
@@ -8,3 +8,4 @@
       %p.lower-message__content
         = message.content
     = image_tag message.image.url, class: 'lower-message__image' if message.image.present?
+  
```
こちらも削除しましょう。

yukri1030 replied:
削除しました

yukri1030 committed:
不要な空行の削除を行なった

# WHAT
不要な空行の削除を行いました

# WHY
可読性を上げるため

shingoteshima commented:
LGTMです！
